### PR TITLE
Expose scraping endpoint for prometheus metrics

### PR DESCRIPTION
**Pull request by GoutamVerma**
In this PR I have made following additions:
- Added dependencies for exposing metrics for Prometheus.
- Added default configuration for kamon.prometheus in `application.conf` (eclair node).
- Update `Monitoring.md`.

### CODE REVIEW

File: docs/Monitoring.md [hunk 1]
```diff
@@ -42,14 +42,61 @@ eclair.sh -with-kanela
...
+kamon.modules {
+  apm-reporter {
+    enabled = false
+  }
+} 
+```
+
+You should then configure your Prometheus process to scrape metrics from the exposed http server. 
+* Download the Prometheus from [here](https://prometheus.io/download/).
+* Configure the `prometheus.yml` file present in prometheus repository and add following in it
```
Comment (t-bast) [inline docs/Monitoring.md:77]:
```suggestion
* Download Prometheus [here](https://prometheus.io/download/).
* Add the following configuration to the `prometheus.yml` file (see the [Prometheus documentation](https://prometheus.io/docs/prometheus/latest/configuration/configuration/) for more details)

```

**Review by t-bast — COMMENTED** [review]
This is starting to look good, I have a few comments, please fix them and once it's done we'll be able to merge this first PR and move on to the next steps.

File: docs/Monitoring.md [hunk 2]
```diff
@@ -42,14 +42,61 @@ eclair.sh -with-kanela
...
+  # The job name is added as a label `job=<job_name>` to any timeseries scraped from this config.
+  - job_name: 'eclair'
+
+    # Override the global default and scrape targets from this job every 5 seconds.
+    scrape_interval: 5s
+
+    static_configs:
+      - targets: ['0.0.0.0:9090']
+```
+After making changes in the prometheus.yml file, start Prometheus and check the status of Prometheus by `sudo systemctl status prometheus`. If it is actively running, you can visit the Prometheus web interface at a specified target in `prometheus.yml`.
```
[the remark this answers is not in the document]
Comment (GoutamVerma) [reply docs/Monitoring.md:98]: Yes, It will be more clear and easier for users.

[the remark this answers is not in the document]
Comment (t-bast) [reply docs/Monitoring.md:98]: That sentence isn't remove in the current version of the PR, did you forget it?

File: docs/Monitoring.md [hunk 3]
```diff
@@ -42,14 +42,61 @@ eclair.sh -with-kanela
...
+scrape_configs:
+  # The job name is added as a label `job=<job_name>` to any timeseries scraped from this config.
+  - job_name: 'eclair'
+
+    # Override the global default and scrape targets from this job every 5 seconds.
+    scrape_interval: 5s
+
+    static_configs:
+      - targets: ['0.0.0.0:9090']
+```
```
[the remark this answers is not in the document]
Comment (t-bast) [reply docs/Monitoring.md:97]: You haven't updated the PR to reflect this?

File: docs/Monitoring.md [hunk 4]
```diff
@@ -30,8 +30,7 @@ kamon {
     sampler = "random"
   }
 
-}
-```
+}```
```
Comment (t-bast) [inline docs/Monitoring.md:33]: Why did you change that? Just look at the result on github and see that it breaks the formatting: https://github.com/GoutamVerma/eclair/blob/prometheus_monitoring/docs/Monitoring.md

Please revert that and fix it line 71 as well.

Comment (GoutamVerma) [reply docs/Monitoring.md]: Now it's looks better
https://github.com/GoutamVerma/eclair/blob/prometheus_monitoring/docs/Monitoring.md

**Comment by t-bast** [comment]
@GoutamVerma now that this first step is complete, you should focus on the following tasks:

- list all metrics that are created and figuring out which ones are most important for monitoring
- create grafana dashboards to monitor those metrics
- open a PR to add the xml for these grafana dashboards

Does that make sense? Start looking into it, and we can do a video call if you have questions.

**Comment by GoutamVerma** [comment]
> * list all metrics that are created and figuring out which ones are most important for monitoring

Yes mentor, I have started looking at essential metrics first for monitoring node. Once I have some  information of grafana dashboard and list of metrics, we will arrange to meet. 

I will shortly get in touch with you.

**Comment by GoutamVerma** [comment]
> Does that make sense? Start looking into it, and we can do a video call if you have questions.

Mentor @t-bast, I have taken a look at metrics available at our Prometheus endpoint and I think we should start making Grafana template in context of our Kamon dashboard. And it will be great if we could arrange a meet call for discussing the stuff😃.

**Comment by t-bast** [comment]
Please spend more time looking into it. There are a lot of metrics, most of which you probably won't understand without understanding more of how lightning works. Make sure you've tried opening channels, sending payments and closing channels between eclair nodes via the API, and trying to figure out what can go wrong and what a node operator would want to monitor. You should spend at least a few days doing this full time, otherwise you'll have no idea what to put inside the dashboards.

**Comment by t-bast** [comment]
Your goal before our next sync should be to understand enough about lightning and eclair to tell me what you think are the important things that should be monitored. You can prepare a list of that based on what you discover playing around with multiple eclair nodes on regtest to simulate a small lightning network.

**Comment by GoutamVerma** [comment]
> Your goal before our next sync should be to understand enough about lightning and eclair to tell me what you think are the important things that should be monitored. You can prepare a list of that based on what you discover playing around with multiple eclair nodes on regtest to simulate a small lightning network.

Yes mentor, I planed to setup two virtual machines(ubuntu) and run both of them parallel. I will open channels, make payments and performs tasks between eclair nodes via API on `regtest`. I will keep updating you with current status.

**Comment by GoutamVerma** [comment]
> Your goal before our next sync should be to understand enough about lightning

Mentor @t-bast, I have done some experiments between two eclair nodes with the help of eclair-cli and performed some major actions between them through this [article](https://acinq.github.io/eclair/#getinfo-2). I also prepared a list of some metrics based on my experiment ([Here](https://docs.google.com/spreadsheets/d/1LuuWWRnGU5Hfll5X5xvYsvIarrAw1aDGhO-m_ht_tZ0/edit?usp=sharing) is attachment). And I want to discuss some more things regarding this work. could we arrange a meeting call in this week at your accommodable time?

File: eclair-core/pom.xml [hunk 5]
```diff
@@ -270,6 +270,11 @@
             <version>1.5.0</version>
         </dependency>
         <!-- MONITORING -->
+        <dependency>
+            <groupId>io.kamon</groupId>
+            <artifactId>kamon-prometheus_${scala.version.short}</artifactId>
+            <version>2.5.4</version>
```
Comment (pm47) [inline eclair-core/pom.xml:276]: This should have used `${kamon.version}`, and also the dependency should have been put in `eclair-node/pom.xml`, (right below `kamon-apm-reporter`), not `eclair-core/pom.xml`

Comment (t-bast) [reply eclair-core/pom.xml:276]: Good catch! @GoutamVerma can you fix that in a new PR?

Comment (GoutamVerma) [reply eclair-core/pom.xml:276]: @pm47 thank you for pointing out, I will fix that in new PR.
